### DISCUSSION TITLE
Enum types for SQLite schemas

### DISCUSSION THREAD

**Discussion by mnpqraven** [Brendonovich/prisma-client-rust#275] [Q&A] [answered]
So since SQLite doesn't support enum types, is there a way to (better) safely cast data from prisma to rust ? This is what I'm currently doing with my schema
e.g
```prisma
model Account
id    String @id @default(cuid())
role  String // supposed to only take User or Admin
```
```rust
enum Role {
  User,
  Admin
}
fn cast_role(role_taken_from_prisma: String) -> Result<Role, CastError> {
  // cast logic and handle errors
}
```

**Comment by Brendonovich** [comment, marked as answer]
Yeah that's about as good as it gets, implementing `FromStr` may be a bit more ergonomic but the same problem still applies.
